webknossos/webknossos/dataset/dataset.py

```diff
@@ -535,6 +556,36 @@ def copy_dataset(
         new_ds._export_as_json()
         return new_ds
 
+    def shallow_copy_dataset(
+        self,
+        new_dataset_path: Path,
+        name: Optional[str] = None,
+        make_relative: bool = False,
+    ) -> "Dataset":
+        """
+        Create a new dataset at the given path. Link all mags of all existing layers.
+        In addition, link all other directories in all layer directories
+        to make this method robust against additional files e.g. layer/mappings/agglomerate_view.hdf5.
+        This method becomes useful when exposing a dataset to webknossos.
+        """
+        new_dataset = Dataset.create(
+            new_dataset_path, scale=self.scale, name=name or self.name
+        )
+        for layer_name, layer in self.layers.items():
+            new_layer = new_dataset.add_layer_like(layer, layer_name)
+            for mag_view in layer.mags.values():
+                new_layer.add_symlink_mag(mag_view, make_relative)
+
+            # copy all other directories with a dir scan
+            copy_directory_with_symlinks(
+                layer.path,
+                new_layer.path,
+                ignore=[str(mag) for mag in layer.mags] + [PROPERTIES_FILE_NAME],
+                make_relative=make_relative,
+            )
```

Comment:
Alternatively you could also link all directories first and use `add_layer_for_existing_files` afterwards, or does this do something different? But no strong opinion.

Reply:
Currently, the links are on the mag level so adding another mag (through downsampling) would not invalidate the source dataset. I could still link on mag level and use the `add_layer_for_existing_files` afterwards, but I like explicit `add_layer_like` and `add_symlink_mag` more.

Reply:
Exactly, I thought also about linking on the mag level, but feel free to leave as-is.